MTFH/Accepted/Integration-with-mmh.md

```diff
@@ -0,0 +1,81 @@
+#  Integration with MMH-derived services (Assets API / Housing Search API)
+
+### **Date:** 15th September 2021
+
+### **Status:** ACCEPTED
+
+## **Context**
+
+The repairs context has evolved naturally in the context of Modern Tools for Housing, it's parent programme, as a separate discrete service. Since the cyber attack, this has been developed in isolation from the other MTFH services, sharing only a data store. The time has come to integrate Repairs with the services previously developed by Manage my Home (formerly Tenants & Leaseholders). We will be starting with two integration points:
+
+1) The Housing Search API during the property search flow
+2) The Assets API during the raise a repair flow
+
+Two options for integration were discussed:
+
+## Option 1: Integrate at the API level with Assets API and Housing Search API
+
+Property search flow:
+1. User enters search screen and enters prop address
+2. API call to property search API
+3. Displayed results as usual with lookup to work order history via propref
+4. User selects the correct address + enters property details page
+5. API call to Assets API with the ID of the clicked record to retrieve full property details
+
+### Advantages:
+- Aligned with Hackney’s current best practice
+- The least effort out of the “do something” options
+- Journey similar to existing journeys in T&L and Finance - use same API approach
+- No data duplication
+- Consolidating the search views is much easier to achieve in future
+- Makes it easier if in the future we need additional property data as assets service is source of truth
+
+### Disadvantages:
+- Tightly coupled services (if Assets is down, RepairsHub is down)
+- Will reduce performance and increase network traffic
+- Driving further down the “API hops” route
+- Lose out on EDA advantages
```

Comment:
We would still be using EDA, but for this specific use case, we want to avoid replicating data and creating one more assets data store - EDA is used for Housing Search and Assets records to maintain them up to date.

Reply:
Clarified that the advantages would be lost for this integration specifically.